examples/elect_leaders.c

```diff
@@ -0,0 +1,320 @@
+
+        argc--;
+        argv++;
+        if (argc > 0) {
+                partitions = rd_kafka_topic_partition_list_new(argc / 2);
+                for (i = 0; i < argc; i += 2) {
+                        rd_kafka_topic_partition_list_add(
+                            partitions, argv[i],
+                            parse_int("partition", argv[i + 1]));
+                }
+        }
+
+        elect_leaders = rd_kafka_ElectLeaders_new(election_type, partitions);
+
+        if (partitions) {
+                rd_kafka_topic_partition_list_destroy(partitions);
+        }
+
+        /*
+         * Create consumer instance
+         * NOTE: rd_kafka_new() takes ownership of the conf object
+         *       and the application must not reference it again after
+         *       this call.
+         */
+        rk = rd_kafka_new(RD_KAFKA_CONSUMER, conf, errstr, sizeof(errstr));
+        if (!rk)
+                fatal("Failed to create new consumer: %s", errstr);
+
+        /*
+         * Elect Leaders
+         */
+        queue = rd_kafka_queue_new(rk);
+
+        /* Signal handler for clean shutdown */
+        signal(SIGINT, stop);
+
+
+        options = rd_kafka_AdminOptions_new(rk, RD_KAFKA_ADMIN_OP_ELECTLEADERS);
+
+        if (rd_kafka_AdminOptions_set_request_timeout(
+                options, 10 * 1000 /* 10s */, errstr, sizeof(errstr))) {
+                fprintf(stderr, "%% Failed to set timeout: %s\n", errstr);
+                goto exit;
+        }
+
+        if (rd_kafka_AdminOptions_set_operation_timeout(
+                options, 10 * 1000 /* 10s */, errstr, sizeof(errstr))) {
+                fprintf(stderr, "%% Failed to set operation timeout: %s\n",
+                        errstr);
+                goto exit;
+        }
+
+        rd_kafka_ElectLeaders(rk, elect_leaders, options, queue);
+
+        rd_kafka_ElectLeaders_destroy(elect_leaders);
+        rd_kafka_AdminOptions_destroy(options);
+
+        /* Wait for results */
+        event = rd_kafka_queue_poll(queue, -1 /* indefinitely but limited by
+                                               * the request timeout set
+                                               * above (10s) */);
+
+        if (!event) {
+                /* User hit Ctrl-C,
+                 * see yield call in stop() signal handler */
+                fprintf(stderr, "%% Cancelled by user\n");
+
+        } else if (rd_kafka_event_error(event)) {
+                rd_kafka_resp_err_t err = rd_kafka_event_error(event);
+                /* ElectLeaders request failed */
+                fprintf(stderr, "%% ElectLeaders failed[%" PRId32 "]: %s\n",
+                        err, rd_kafka_event_error_string(event));
+                goto exit;
+        } else {
+                /* ElectLeaders request succeeded */
+                const rd_kafka_ElectLeaders_result_t *result;
+                result = rd_kafka_event_ElectLeaders_result(event);
+                retval = print_elect_leaders_result(result);
```

Comment:
When returning void, retval isn't to be modified
```suggestion
                print_elect_leaders_result(result);
```